app/charts/shared/legend-color.tsx

```diff
@@ -190,45 +199,52 @@ export const LegendColor = memo(function LegendColor(props: LegendColorProps) {
 
 const removeOpacity = (rgb: number[]) => rgb.slice(0, 3);
 
-// TODO: add metadata to legend titles?
-export const MapLegendColor = memo(function LegendColor({
-  component,
-  getColor,
-  useAbbreviations,
-  chartConfig,
-}: {
+type MapLegendColorProps = {
   component: Component;
   getColor: (d: Observation) => number[];
   useAbbreviations: boolean;
   chartConfig: MapConfig;
-}) {
+  observations: Observation[];
+};
+
+export const MapLegendColor = React.memo(function LegendColor(
+  props: MapLegendColorProps
+) {
+  const { component, getColor, useAbbreviations, chartConfig, observations } =
+    props;
   const filters = useChartConfigFilters(chartConfig);
-  const componentFilter = filters[component.iri];
-  const sortedValues = useMemo(() => {
+  const dimensionFilter = filters[component.iri];
+  const sortedValues = React.useMemo(() => {
     const sorters = makeDimensionValueSorters(component, {
-      sorting: {
-        sortingType: "byAuto",
-        sortingOrder: "asc",
-      },
-      dimensionFilter: componentFilter,
+      sorting: { sortingType: "byAuto", sortingOrder: "asc" },
+      dimensionFilter,
     });
     return orderBy(
       component.values,
-      sorters.map((s) => (dv) => s(dv.label))
-    ) as typeof component.values;
-  }, [component, componentFilter]);
-  const getLabel = useAbbreviations
-    ? (d: string) => {
+      sorters.map((s) => (d) => s(d.label))
+    );
+  }, [component, dimensionFilter]);
+  const getLabel: (d: string) => string = useAbbreviations
+    ? (d) => {
         const v = component.values.find((v) => v.value === d);
         return (v?.alternateName || v?.label) as string;
       }
-    : (d: string) => {
-        return component.values.find((v) => v.value === d)?.label as string;
-      };
+    : (d) => component.values.find((v) => v.value === d)?.label as string;
+  const observationLabels = React.useMemo(() => {
+    return [...new Set(observations.map((o) => o[component.iri]))];
+  }, [observations, component.iri]);
+  const sortedStringValues = sortedValues.map((d) => `${d.value}`);
   const groups = useLegendGroups({
     chartConfig,
     title: component.label,
-    values: sortedValues.map((d) => `${d.value}`),
+    values:
+      // If the component is a measure or an ordinal dimension, we want to show all
+      // values for comparison purposes
+      isOrdinalDimension(component) || isOrdinalMeasure(component)
+        ? sortedStringValues
+        : sortedStringValues.filter((v) =>
+            observationLabels.includes(getLabel(v))
```

Comment:
observationLabels could be a Set, so that we can use has instead of includes. Should be more efficient (O(n) vs O(1))

Reply:
I think it would be a bit clearer to have legendValues inside it's own memo. sortedStringValues could be computed inside the memo.

Reply:
Interesting, I wasn't aware that there's such a performance difference between set and array 💡